source/schedulexmlnode.d

```diff
@@ -0,0 +1,249 @@
+module scheduleXmlNode;
+
+import std.datetime;
+import kxml.xml : XmlNode;
+import std.array : empty, front;
+
+import xmlconstants;
+import xmlparserexceptions;
+
+public: 
+struct ScheduleXmlNode {
+    // The innerNode should not be writeable, but alias this needs access.
+    private XmlNode _innerNode;
+    public XmlNode innerNode() @property {
+        return _innerNode;
+    }
+    alias innerNode this;
+
+    public this(XmlNode node) {
+        this._innerNode = node;
+    }
+
+    public auto departureDate(string _dateNodeName = dateNodeName)()
+    in
+    {
+        assert(getName == departureNodeName);
+    }
+    body
```

Comment:
We want to use the new `do` syntax instead of `body`, so if we touch the code anyways, just do it at that moment